### PR TITLE
Add a cache to load complementarity constraints in MOI wrapper

### PR DESCRIPTION
**Pull request by frapac**
This PR allows to load MPEC constraints sequentially in the MOI wrapper (see #169). That allows more flexibility when we are using JuMP to define the MPEC constraints. For instance, with this PR, we could load complementarity constraints sequentially, as
```julia
@constraint(model, [M*x + b; x[3:5]] in MOI.Complements(3))
@constraint(model, [N*y + d; y[3:5]] in MOI.Complements(3))
```
which was forbidden previously. 

However, we could not define two complementarity constraints sharing common variables. E.g. the following code will return an error in Knitro:
```julia
@constraint(model, [x[1]; x[2]] in MOI.Complements(1))
@constraint(model, [x[1]; x[3]] in MOI.Complements(1))
```
A solution would be to add a new  variable `x_tmp` in the cache, and adding a new constraint `x_tmp = x[1]` to define properly the second complementarity constraint as `[x_tmp; x[3]] in MOI.Complements(1)`. 

So to sum it up, the question is: how many operations should we implement in `ComplementarityCache` to improve the user experience when defining MPEC with JuMP?

### CODE REVIEW

**Comment by joaquimg** [comment]
Excellent question, I missed it earlier.
I vote for adding the new variables.
For instance, Xpress tracks the positiveness of some variables in SOCs.

However, I would consider merging this as is and add more functionality later. This is already a nice upgrade on the user experience.

Maybe @odow has some input.

**Comment by odow** [comment]
I think just throw an error. It doesn't really make sense to have multiple complementary constraints on the same variable. Perhaps in the error explain the work-around.

**Comment by joaquimg** [comment]
good for me!
